core/station/impl/src/repositories/indexes/request_policy_resource_index.rs

```diff
@@ -354,21 +386,39 @@ mod tests {
 
             repository.insert(index);
 
-            if i % 2 == 0 {
+            if i % 2 != 0 {
                 expected_method_ids.push([i; 16]);
             }
         }
 
         expected_method_ids.reverse();
 
-        for i in (0..20).step_by(2) {
+        for i in (1..20).step_by(2) {
```

Comment:
Now we're losing coverage for cases of no validation method.

Also it'd be useful to test the case of two policies with the same execution method and (i) no validation, (ii) some validation while
- retrieving by execution method (expected to get them both)
- retrieving by execution and no validation method (expected to get one of them)
- retrieving by execution and some validation method (expected to get the other one of them)

Reply:
Good suggestion, I’ll extend the test tomorrow